plasma_framework/docs/design/exit_bond_and_bounty_mechanics.md

```diff
@@ -0,0 +1,189 @@
+# Exit Bond and Bounty Mechanics
+
+This document is a description of the current implementation of exit bonds and process exit bounties which are an integral part of the exit mechanism.
+
+A Process Exit Bounty is a reward which is paid out to the `processExit()` initiator to compensate for the transaction cost. The motive of having a bounty is to remove an extra step of processing exits while exiting funds. This also ensures everyone pays their share of processing contrary to passing on the responsibility to someone else.
+
+The Process Exit Bounty is paid out as a portion (or whole) of the Exit Bond and the size of the Bond is designed to ideally cover both the cost of challenging an invalid exit or the cost of processing a valid exit 
+
+## The Bond Life Cycle
+
+### For Standard Exits
+
+The Bond is supplied by the Output owner while starting an exit and the main motive is to disincentivize invalid exits while also being able to use a portion of the bond to incentivize processing exits. The Bond is internally divided into two portions. For successful exits, one of them can be returned to the output owner, and the other is a reward for processing the exit.
+The initiator/owner has to obtaint the correct size of the Exit Bond by calling `startStandardExitBondSize()`.
+
+**_After Starting an Exit:_**
+*Bond -> from `Exit owner` to `Exit Game Contract`*
+
+For an invalid exit, a challenge will transfer the whole bond to the challenger. Both exitable and non-exitable exits still remain in the exit queue and is deleted when the exits are processed. For a scenario where there are mutliple invalid exits in the queue, the processor is still awarded with some gas reward for deleting non-exitable exits from storage.
+
+**_On succesfull challenge:_**
+*Bond -> from `Exit Game Contract` to `Challenger`*
+
+The user who processes an exit, gets the bounty reward for processing an exit. The cost of processing multiple exits from the exit queue at once could be of less cost than processing all those exits individually. However, the bounty amount associated with each exit is given out to the processor.
+
+**_On processing exit:_**
+*Bond - Bounty -> from `Exit Game Contract` to `Exit owner`*\
+*Bounty -> from `Exit Game Contract` to `Exit processor`*
+
+### For In-Flght Exits
+
+The Bond/Bounty collection for in-flight exits is similar. However, for an in-flight exit, the Exit Bounty is a poriton of the Piggyback Bond. Since only the piggybacked input/output from an in-flight exit can be exited, it only makes sense to have the bounty for each piggyback from the respective owner. The size of the Piggyback Bond can be obtained by calling `piggybackBondSize()`.
```

Comment:
```suggestion
The situation for in-flight exits is slightly different. Since the piggybacked inputs or outputs of an in-flight exit are exited separately, the **Exit Initiator** is the owner of the piggybacked input or output. In this case the **Exit Bond** is the Piggyback Bond that is put up when piggybacking and the **Process Exit Bounty** is taken from this. The size of the Piggyback Bond can be obtained by calling `piggybackBondSize()`.
```